interoperator/controllers/multiclusterdeploy/sfserviceinstancereplicator/sfserviceinstancereplicator_controller.go

```diff
@@ -383,7 +382,23 @@ func (r *InstanceReplicator) reconcileServicePlan(targetClient client.Client, in
 	if err != nil {
 		log.Error(err, "Failed to get SFPlan from leader")
 		lastErr = err
-	}
+	} else if deleted && !plan.GetDeletionTimestamp().IsZero() {
+        annotations := plan.GetAnnotations()
+        if annotations == nil {
+            annotations = make(map[string]string)
+        }
+        deleteAttempts, ok := annotations[constants.PlanDeleteAttempts];
+        switch {
```

Comment:
I think it will be helpful if we add comments here on why we are updating the plan annotation.

Reply:
done.. thanks.